src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509AuthorityKeyIdentifierExtension.cs

```diff
@@ -0,0 +1,575 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Formats.Asn1;
+using System.Runtime.InteropServices.ComTypes;
+using System.Security.Cryptography.Asn1;
+using System.Security.Cryptography.X509Certificates.Asn1;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    /// <summary>
+    ///   Represents the Authority Key Identifier X.509 Extension (2.5.29.35).
+    /// </summary>
+    public sealed class X509AuthorityKeyIdentifierExtension : X509Extension
+    {
+        private bool _decoded;
+        private X500DistinguishedName? _simpleIssuer;
+        private ReadOnlyMemory<byte>? _keyIdentifier;
+        private ReadOnlyMemory<byte>? _rawIssuer;
+        private ReadOnlyMemory<byte>? _serialNumber;
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="X509AuthorityKeyIdentifierExtension" />
+        ///   class.
+        /// </summary>
+        public X509AuthorityKeyIdentifierExtension()
+            : base(Oids.AuthorityKeyIdentifier)
+        {
+            _decoded = true;
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="X509AuthorityKeyIdentifierExtension" />
+        ///   class from an encoded representation of the extension and an optional critical marker.
+        /// </summary>
+        /// <param name="rawData">
+        ///   The encoded data used to create the extension.
+        /// </param>
+        /// <param name="critical">
+        ///   <see langword="true" /> if the extension is critical;
```

Comment:
"critical" has a domain-specific meaning here that a user of this API will understand?

Reply:
> "critical" has a domain-specific meaning here?

Yep.

> ... that a user of this API will understand?

That's debatable; but it's the same as with all of the other X509Extension types.